packages/devtools/src/sourceMap/loadOriginalSourceLoc.ts

```diff
@@ -0,0 +1,178 @@
+import type { MappedPosition, RawSourceMap } from 'source-map-js';
+import { getFilePath, getOriginalPosition, resources } from './sourceMapConsumer';
+
+// when only one source map appears in a source file: source url -> source map JSON
+// when more than one source maps appear in a source file: `${source}:${line}:${column}` -> source map JSON
+const sourceMapJSONs: Map<string, RawSourceMap> = new Map();
+
+const getComputedSouceMapJSON = ({ source, line, column }: MappedPosition) => {
+  return sourceMapJSONs.get(source) ?? sourceMapJSONs.get(`${source}:${line}:${column}`);
+};
+
+export async function loadOriginalSourceLoc(sourceUrlWithLoc: string): Promise<MappedPosition> {
+  const paths = sourceUrlWithLoc.split(':');
+  const column = Number(paths.pop());
+  const line = Number(paths.pop());
+  const source = paths.join(':');
+  if (Number.isNaN(line) || Number.isNaN(column)) {
+    return { source, line: 1, column: 0 };
+  }
```

Comment:
```suggestion
  const paths = sourceUrlWithLoc.split(':');
  const column = Number(paths.pop());
  const line = Number(paths.pop());
  const source = paths.join(':');
  if (Number.isNaN(line) || Number.isNaN(column)) {
    return { source, line: 1, column: 0 };
  }
```

nit: you could extract this to a `parseSoureUrl` function

```ts
function parseSourceUrl(sourceUrlWithLoc: string): MappedPosition {
  const paths = sourceUrlWithLoc.split(':');
  const column = Number(paths.pop());
  const line = Number(paths.pop());
  const source = paths.join(':');
  if (Number.isNaN(line) || Number.isNaN(column)) {
    return { source, line: 1, column: 0 };
  }

  return { source, line, column };
}
```